### PR TITLE
Remove deprecated AWS, GCE terraform configurations

### LINKED ISSUE #1693: remove (or deprecate) legacy terraform
We should consider removing or at least clearly deprecating the legacy terraform configuration for AWS and GCE. They are a maintenance burden and are causing confusion.

Legacy:

./terraform/aws/aws.tf
./terraform/gce/gce.tf

### PR DESCRIPTION
**Pull request by siddharthist**
- [x] Installs cleanly on a fresh build of most recent master branch
- [x] Upgrades cleanly from the most recent release
- [x] Updates documentation relevant to the changes

Also switches Travis to the new modules.

Waiting on CI build.

fixes #1693, #1692

### CODE REVIEW

File: testing/terraform/aws.tf [hunk 1]
```diff
@@ -1,19 +1,142 @@
...
+}
+variable "availability_zones"  {
+  default = "a,b,c"
+}
+variable "control_count" { default = 3 }
+variable "datacenter" {default = "aws-us-west-2"}
+variable "edge_count" { default = 1 }
+variable "region" {default = "us-west-1"}
+variable "short_name" {default = "mantl-ci-${var.build_number}"}
+variable "long_name" {default = "mantl-ci-${var.build_number}"}
```
Comment (ryane) [inline testing/terraform/aws.tf]: looks like you just need to set these values outside of the variable definition

Comment (siddharthist) [inline testing/terraform/aws.tf]: Oh yeah, I forgot you can use variables in defaults. Thanks for the pointer! <-- still a newbie to terraform

**Comment by siddharthist** [comment]
This may have to wait until we update our AWS configuration to work with Terraform 0.7. I can `terraform plan` manually with Terraform 0.6.15.

**Comment by siddharthist** [comment]
This depends on https://github.com/CiscoCloud/mantl/pull/1785

**Comment by thomasvincent** [comment]
@Theaxiom

**Pull request merged**